semantics/executable-spec/src/Data/Coders.hs

```diff
@@ -237,8 +240,18 @@ decodeCollectionWithLen lenOrIndef el = do
 encodeFoldable :: (ToCBOR a, Foldable f) => f a -> Encoding
 encodeFoldable = encodeFoldableEncoder toCBOR
 
+encodeFoldableMapPairs :: (ToCBOR a, ToCBOR b, Foldable f) => f (a,b) -> Encoding
+encodeFoldableMapPairs = encodeFoldableEncoderAs wrapCBORMap $
+  \(a,b) -> toCBOR a <> toCBOR b
+
 encodeFoldableEncoder :: (Foldable f) => (a -> Encoding) -> f a -> Encoding
-encodeFoldableEncoder encoder xs = wrapCBORArray len contents
+encodeFoldableEncoder = encodeFoldableEncoderAs wrapCBORArray
+
+encodeFoldableEncoderAs :: (Foldable f) =>
+  (Word -> Encoding -> Encoding) ->
+  (a -> Encoding) ->
+   f a -> Encoding
+encodeFoldableEncoderAs wrap encoder xs = wrap len contents
```

Comment:
Some comments about how this is intended to be used.

Reply:
This isn't a new function being exported. This is common functionality factored out between `encodeFoldableEncoder` and the (new) `encodeFoldableMapPairs`